### PR TITLE
Refactor `BlockPayload` to a struct

### PR DESCRIPTION
**Pull request by shenkeyao**
* Refactors the `BlockPayload` trait to a struct and removes `VIDBlockPayload`.
* Replaces the `Payload` associated types with `Transaction` for `NodeImplementation`, `State`, and `BlockHeader` traits.

Closes https://github.com/EspressoSystems/HotShot/issues/1815.

### CODE REVIEW

File: crates/hotshot/src/demo.rs [hunk 1]
```diff
@@ -106,7 +104,7 @@ impl TestableState for DemoState {
         _state: Option<&Self>,
         _rng: &mut dyn rand::RngCore,
         padding: u64,
-    ) -> <Self::BlockPayload as BlockPayload>::Transaction {
+    ) -> VIDTransaction {
```
Comment (rob-maron) [inline crates/hotshot/src/demo.rs:107]: Love it!

**Review by rob-maron — APPROVED** [review]
Looks good to me !

File: crates/hotshot/src/demo.rs (hunk 1 above)
Comment (shenkeyao) [reply crates/hotshot/src/demo.rs:107]: Thanks for the review! I'm actually going to close this PR since we've decided to keep the payload as a trait.

**Comment by shenkeyao** [comment]
Closing this PR without merging to keep the payload trait.

**Pull request closed without merging**